pytorch_lightning/loops/optimization/closure.py

```diff
@@ -188,3 +189,25 @@ def closure(self, *args: Any, **kwargs: Any) -> ClosureResult:
                     step_output.closure_loss = self._backward_fn(step_output.closure_loss)
 
         return step_output
+
+
+class _FaultToleranceClosureExecutor:
+
+    """This class is used to prevent fault tolerant to create a checkpoint while parameters are being updated."""
+
+    def __init__(self, closure: Closure, trainer: "pl.Trainer"):
+        self._closure = closure
+        self._trainer = trainer
+        self._fault_tolerant_possible: Optional[bool] = None
+
+    def consume_result(self) -> ClosureResult:
+        return self._closure.consume_result()
+
+    def __call__(self):
+        # enable fault tolerant during closure execution
+        with self._trainer._fault_tolerant_supported(enable=True):
+            loss = self._closure()
+        self._fault_tolerant_possible = self._trainer._fault_tolerant_possible
+        # prevent fault tolerant during parameters update.
+        self._trainer._fault_tolerant_possible = False
```

Comment:
do we expect a failure during the parameter update in the optimizer?

Reply:
don't we want the lightning optimizer to take care of this?

Reply:
The LightningOptimizer can't handle this logic as the closure is being passed the accelerators. The only solution I found was to wrap the closure on optimizer.step call.